vsintegration/src/FSharp.Editor/DocComments/XMLDocumentation.fs

```diff
@@ -215,10 +214,9 @@ module internal XmlDocumentation =
     /// Provide Xml Documentation             
     type Provider(xmlIndexService:IVsXMLMemberIndexService) = 
         /// Index of assembly name to xml member index.
-        let mutable xmlCache = new AgedLookup<VsThreadToken,string,IVsXMLMemberIndex>(10,areSimilar=(fun (x,y) -> x = y))
+        let cache = Dictionary<string, IVsXMLMemberIndex>()
```

Comment:
Does this need to be a ConcurrentDictionary?

Reply:
I don't think so. At least the previous data structure isn't a concurrent one, and AFAIK it's not really a scenario where multiple threads are calling into this routine concurrently.

Reply:
@cartermp it's probably worth taking a lock, or using concurrent dictionary.  We aren't actively requiring ui thread only, so multi threaded use of the IDocumentionBuilder will eventually occur.

Reply:
@cartermp the aged lookup is a weak reference fix sized cache.  Dictionary is not, I assume we believe this won't cause a problem, because the lifetime of this dictionary is quite short.

Reply:
I'll look into that

Reply:
Note that there doesn't seem to be anything about AgedLookup that helps a multi-threaded scenario, so really the difference here is:

* O(1) lookup time instead of O(n), which AgedLookup has
* Strongly-held data rather than a max of 75 items at a given time

Reply:
I think using a concurrent data structure here is probably not the right call here at all. The service isn't concurrent as far as I can tell, as this cache exists inside a type that side inside a `ConditionalWeakTable` bound to `documentationBuilderCache`, a module-level value. I think if we were multi-threaded then making this specific cache a concurrent dictionary would be the least of our problem